File: barcode-creator-framework-console-cs/LinearBarcode.cs
using ceTe.DynamicBarcode.Creator;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace barcode_creator_framework_console_cs
{
    class LinearBarcode
    {
        public static void CodaBarBarcode(string outputPath)
        {
            Codabar barcode = new Codabar("A1234A", 150);
            barcode.Draw(outputPath, 100, ImageFormat.Png);
        }

        public static void Code11Barcode(string outputPath)
        {
            Code11 barcode = new Code11("12345678", 50);
            barcode.BarcodeMargin = new Margin(2);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Code128Barcode(string outputPath)
        {
            Code128 barcode = new Code128("Hello World", 100);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Code25Barcode(string outputPath)
        {
            Code25 barcode = new Code25("123456789", 100);
            barcode.ShowText = true;
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Code39Barcode(string outputPath)
        {
            Code39 barcode = new Code39("123456789", 100);
            barcode.SymbolHeight = 12;
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Code93Barcode(string outputPath)
        {
            Code93 barcode = new Code93("1234567890", 50);
            barcode.ShowText = true;
            barcode.TextAlign = Align.Right;
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean13Barcode(string outputPath)
        {
            Ean13 barcode = new Ean13("123456789012");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean13Supplement2Barcode(string outputPath)
        {
            Ean13Supplement2 barcode = new Ean13Supplement2("123456789012", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean13Supplement5Barcode(string outputPath)
        {
            Ean13Supplement5 barcode = new Ean13Supplement5("123456789012", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean14Barcode(string outputPath)
        {
            Ean14 barcode = new Ean14("1234567890123", 100);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean8Barcode(string outputPath)
        {
            Ean8 barcode = new Ean8("12345670");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean8Supplement2Barcode(string outputPath)
        {
            Ean8Supplement2 barcode = new Ean8Supplement2("12345670", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Ean8Supplement5Barcode(string outputPath)
        {
            Ean8Supplement5 barcode = new Ean8Supplement5("12345670", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void GS1DataBarBarcode(string outputPath)
        {
            GS1DataBar barcode = new GS1DataBar("(01)9889876543210", 100, GS1DataBarType.Omnidirectional);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Iata25Barcode(string outputPath)
        {
            Iata25 barcode = new Iata25("1234567890", 100);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Interleaved25Barcode(string outputPath)
        {
            Interleaved25 barcode = new Interleaved25("1234567890", 100);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IsbnBarcode(string outputPath)
        {
            Isbn barcode = new Isbn("978-1-23-456789-7");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IsbnSupplement2Barcode(string outputPath)
        {
            IsbnSupplement2 barcode = new IsbnSupplement2("978-1-23-456789-7", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }

        public static void IsbnSupplement5Barcode(string outputPath)
        {
            IsbnSupplement5 barcode = new IsbnSupplement5("978-1-23-456789-7", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IsmnBarcode(string outputPath)
        {
            Ismn barcode = new Ismn("979-0-1234-5678");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IsmnSupplement2Barcode(string outputPath)
        {
            IsmnSupplement2 barcode = new IsmnSupplement2("979-0-1234-5678", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IsmnSupplement5Barcode(string outputPath)
        {
            IsmnSupplement5 barcode = new IsmnSupplement5("979-0-1234-5678", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IssnBarcode(string outputPath)
        {
            Issn barcode = new Issn("977-1234-56700");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IssnSupplement2Barcode(string outputPath)
        {
            IssnSupplement2 barcode = new IssnSupplement2("977-1234-56700", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void IssnSupplement5Barcode(string outputPath)
        {
            IssnSupplement5 barcode = new IssnSupplement5("977-1234-56700", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void Itf14Barcode(string outputPath)
        {
            Itf14 barcode = new Itf14("1234567890123", 100);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }

        public static void MsiBarcodeBarcode(string outputPath)
        {
            MsiBarcode barcode = new MsiBarcode("123456789", 100);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }

        public static void StackedGS1DatabarBarcode(string outputPath)
        {
            StackedGS1DataBar barcode = new StackedGS1DataBar("(01)9889876543210", 50, StackedGS1DataBarType.StackedOmnidirectional);
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void UpcABarcode(string outputPath)
        {
            UpcA barcode = new UpcA("12345678901");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void UpcASupplement5Barcode(string outputPath)
        {
            UpcASupplement5 barcode = new UpcASupplement5("12345678901", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void UpcASupplement2Barcode(string outputPath)
        {
            UpcASupplement2 barcode = new UpcASupplement2("12345678901", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void UpcEBarcode(string outputPath)
        {
            UpcE barcode = new UpcE("0123456");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void UpcESupplement5Barcode(string outputPath)
        {
            UpcESupplement5 barcode = new UpcESupplement5("0123456", "12345");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
        public static void UpcESupplement2Barcode(string outputPath)
        {
            UpcESupplement2 barcode = new UpcESupplement2("0123456", "12");
            barcode.Draw(outputPath, 300, ImageFormat.Png);
        }
    }
}

File: barcode-creator-core-console-vb/LinearBarcode.vb
Imports ceTe.DynamicBarcode.Creator

Public Class LinearBarcode
    Public Shared Sub CodaBarBarcode(ByVal outputPath As String)
        Dim MyBarCode As Codabar = New Codabar("A1234A", 150)
        MyBarCode.BackColor = New ceTe.DynamicBarcode.Creator.Rgb24(0, 191, 255)
        MyBarCode.Draw(outputPath, 100, ImageFormat.Png)
    End Sub

    Public Shared Sub Code11Barcode(ByVal outputPath As String)
        Dim MyBarCode As Code11 = New Code11("12345678", 50)
        MyBarCode.BarcodeMargin = New Margin(2)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Code128Barcode(ByVal outputPath As String)
        Dim MyBarCode As Code128 = New Code128("Hello World", 100)
        MyBarCode.BarColor = New ceTe.DynamicBarcode.Creator.Rgba32(170, 80, 0, 3)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Code25Barcode(ByVal outputPath As String)
        Dim MyBarCode As Code25 = New Code25("123456789", 100)
        MyBarCode.ShowText = True
        MyBarCode.FontSize = 2
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Code39Barcode(ByVal outputPath As String)
        Dim MyBarCode As Code39 = New Code39("123456789", 100)
        MyBarCode.SymbolHeight = 12
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Code93Barcode(ByVal outputPath As String)
        Dim MyBarCode As Code93 = New Code93("1234567890", 50)
        MyBarCode.ShowText = True
        MyBarCode.TextAlign = Align.Right
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean13Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean13 = New Ean13("123456789012")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean13Supplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean13Supplement2 = New Ean13Supplement2("123456789012", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean13Supplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean13Supplement5 = New Ean13Supplement5("123456789012", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean14Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean14 = New Ean14("1234567890123", 100)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean8Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean8 = New Ean8("12345670")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean8Supplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean8Supplement2 = New Ean8Supplement2("12345670", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Ean8Supplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As Ean8Supplement5 = New Ean8Supplement5("12345670", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub GS1DataBarBarcode(ByVal outputPath As String)
        Dim MyBarCode As GS1DataBar = New GS1DataBar("(01)9889876543210", 100, GS1DataBarType.Omnidirectional)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Iata25Barcode(ByVal outputPath As String)
        Dim MyBarCode As Iata25 = New Iata25("1234567890", 100)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Interleaved25Barcode(ByVal outputPath As String)
        Dim MyBarCode As Interleaved25 = New Interleaved25("1234567890", 100)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IsbnBarcode(ByVal outputPath As String)
        Dim MyBarCode As Isbn = New Isbn("978-1-23-456789-7")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IsbnSupplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As IsbnSupplement2 = New IsbnSupplement2("978-1-23-456789-7", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IsbnSupplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As IsbnSupplement5 = New IsbnSupplement5("978-1-23-456789-7", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IsmnBarcode(ByVal outputPath As String)
        Dim MyBarCode As Ismn = New Ismn("979-0-1234-5678")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IsmnSupplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As IsmnSupplement2 = New IsmnSupplement2("979-0-1234-5678", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IsmnSupplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As IsmnSupplement5 = New IsmnSupplement5("979-0-1234-5678", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IssnBarcode(ByVal outputPath As String)
        Dim MyBarCode As Issn = New Issn("977-1234-56700")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IssnSupplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As IssnSupplement2 = New IssnSupplement2("977-1234-56700", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub IssnSupplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As IssnSupplement5 = New IssnSupplement5("977-1234-56700", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub Itf14Barcode(ByVal outputPath As String)
        Dim MyBarCode As Itf14 = New Itf14("1234567890123", 100)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub MsiBarcodeBarcode(ByVal outputPath As String)
        Dim MyBarCode As MsiBarcode = New MsiBarcode("123456789", 100)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub StackedGS1DatabarBarcode(ByVal outputPath As String)
        Dim MyBarCode As StackedGS1DataBar = New StackedGS1DataBar("(01)9889876543210", 50, StackedGS1DataBarType.StackedOmnidirectional)
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub UpcABarcode(ByVal outputPath As String)
        Dim MyBarCode As UpcA = New UpcA("12345678901")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub UpcASupplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As UpcASupplement5 = New UpcASupplement5("12345678901", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub UpcASupplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As UpcASupplement2 = New UpcASupplement2("12345678901", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub UpcEBarcode(ByVal outputPath As String)
        Dim MyBarCode As UpcE = New UpcE("0123456")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub UpcESupplement5Barcode(ByVal outputPath As String)
        Dim MyBarCode As UpcESupplement5 = New UpcESupplement5("0123456", "12345")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub

    Public Shared Sub UpcESupplement2Barcode(ByVal outputPath As String)
        Dim MyBarCode As UpcESupplement2 = New UpcESupplement2("0123456", "12")
        MyBarCode.Draw(outputPath, 300, ImageFormat.Png)
    End Sub
End Class
